internal/updater/utils/html/parsing.go

```diff
@@ -0,0 +1,70 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/andybalholm/cascadia"
+	"golang.org/x/net/html"
+)
+
+var (
+	ErrAttrNotFound = errors.New("matching attribute not found")
+)
+
+func GetText(n *html.Node) string {
+	return n.FirstChild.Data
+}
+
+func GetAttr(n *html.Node, key string) (string, error) {
+	for _, attr := range n.Attr {
+		if attr.Key == key {
+			return attr.Val, nil
+		}
+	}
+
+	return "", ErrAttrNotFound
+}
+
+func CheckID(n *html.Node, id string) bool {
+	if n.Type == html.ElementNode {
+		s, err := GetAttr(n, "id")
+
+		if err == nil && s == id {
+			return true
+		}
+	}
+
+	return false
+}
+
+func Traverse(n *html.Node, id string) *html.Node {
+	if CheckID(n, id) {
+		return n
+	}
+
+	for c := n.FirstChild; c != nil; c = c.NextSibling {
+		res := Traverse(c, id)
+
+		if res != nil {
+			return res
+		}
+	}
+
+	return nil
+}
+
+func GetNodeByID(n *html.Node, id string) *html.Node {
+	return Traverse(n, id)
+}
+
+func GetNodesBySelector(n *html.Node, cssSelector string) []*html.Node {
+	cascSelector := cascadia.MustCompile(cssSelector)
+	cascMatcher := cascadia.Matcher(cascSelector)
+	return cascadia.QueryAll(n, cascMatcher)
+}
+
+func GetFirstNodeBySelector(n *html.Node, cssSelector string) *html.Node {
+	cascSelector := cascadia.MustCompile(cssSelector)
+	cascMatcher := cascadia.Matcher(cascSelector)
+	return cascadia.Query(n, cascMatcher)
+}
```

Comment:
I believe this is unused, let's remove it if so

```suggestion
```

Reply:
This was put in more as an additional HTML utility for future scraping - for anyone who wants to scrape the first matching node by a selector, rather than finding every single matching element. Obviously this could be done with `GetNodesBySelector`, and picking the first element - but this specifically stops hunting through the HTML once the first match is found.

Essentially, I'm trying to replicate some of the common features found in other HTML scrapers - building a small library of tools to try and make grabbing data from other providers easier.